msglibrary/src/test/java/ingsw/group1/msglibrary/SMSPeerTest.java

```diff
@@ -18,113 +15,72 @@
  * @author Giorgia Bortoletti
  * @author Riccardo De Zen
  */
-@Config(sdk = Build.VERSION_CODES.P)
-@RunWith(RobolectricTestRunner.class)
+@RunWith(Parameterized.class)
 public class SMSPeerTest {
 
-    private SMSPeer peer;
-    private static final String EX_VALID_ADDRESS = "+39892424";
-
-    @Test
-    public void isAddressValidNull(){
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_NOT_PHONE_NUMBER,
-                SMSPeer.isAddressValid(null)
-        );
-    }
-
-    @Test
-    public void isAddressValidEmptyOrBlank(){
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_NOT_PHONE_NUMBER,
-                SMSPeer.isAddressValid("")
-        );
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_NOT_PHONE_NUMBER,
-                SMSPeer.isAddressValid("     ")
-        );
-    }
-
-    @Test
-    public void isAddressValidNaN(){
-        String notANumber = "Wassup";
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_NOT_PHONE_NUMBER,
-                SMSPeer.isAddressValid(notANumber)
-        );
-    }
-
-    @Test
-    public void isAddressValidNoCountryCode(){
-        String noCC = "911";
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_NO_COUNTRY_CODE,
-                SMSPeer.isAddressValid(noCC)
-        );
-    }
+    private static final int TEST_RUNS = 100;
+    private static final RandomSMSPeerGenerator GENERATOR = new RandomSMSPeerGenerator();
 
-    @Test
-    public void isAddressValidTooLong(){
-        String longAddress = "+11111111111111111111111111";
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_TOO_LONG,
-                SMSPeer.isAddressValid(longAddress)
-        );
-    }
+    private SMSPeer peer;
 
-    @Test
-    public void isAddressValidTooShort(){
-        String shortAddress = "+10";
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_TOO_SHORT,
-                SMSPeer.isAddressValid(shortAddress)
-        );
+    /**
+     * @return parameters for the test. List is empty because it's needed to simply run the test
+     * multiple times.
+     */
+    @Parameterized.Parameters
+    public static Object[][] data() {
+        return new Object[TEST_RUNS][0];
     }
 
-    @Test
-    public void isAddressValidIsValid(){
-        assertEquals(
-                SMSPeer.PhoneNumberValidity.ADDRESS_VALID,
-                SMSPeer.isAddressValid(EX_VALID_ADDRESS)
-        );
+    /**
+     * Default constructor. Declaration in parameterized tests is needed even if empty.
+     */
+    public SMSPeerTest() {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Avoid unnecessary constructors - the compiler will generate these for you](https://app.codacy.com/manual/RicDeZen/ingsw-group1-message-library/pullRequest?prid=4741620)